plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/expression/RewriteStringComparison.java

```diff
@@ -0,0 +1,117 @@
+    private static final Capture<ConnectorExpression> RIGHT = newCapture();
+    private static final Pattern<Call> PATTERN = call()
+            .with(type().equalTo(BOOLEAN))
+            .with(functionName().matching(Stream.of(ComparisonOperator.values())
+                    .filter(comparison -> comparison != ComparisonOperator.IDENTICAL)
+                    .map(ComparisonOperator::getFunctionName)
+                    .collect(toImmutableSet())
+                    ::contains))
+            .with(argumentCount().equalTo(2))
+            .with(argument(0).matching(expression().with(type().matching(type -> type instanceof CharType || type instanceof VarcharType)).capturedAs(LEFT)))
+            .with(argument(1).matching(expression().with(type().matching(type -> type instanceof CharType || type instanceof VarcharType)).capturedAs(RIGHT)));
+
+    @Override
+    public Pattern<Call> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    public Optional<ParameterizedExpression> rewrite(Call expression, Captures captures, RewriteContext<ParameterizedExpression> context)
+    {
+        ComparisonOperator comparison = ComparisonOperator.forFunctionName(expression.getFunctionName());
+        ConnectorExpression leftExpression = captures.get(LEFT);
+        ConnectorExpression rightExpression = captures.get(RIGHT);
+
+        if (leftExpression instanceof Variable variable && !isString(variable, context)) {
+            return Optional.empty();
+        }
+
+        if (rightExpression instanceof Variable variable && !isString(variable, context)) {
+            return Optional.empty();
+        }
+
+        Optional<ParameterizedExpression> left = context.defaultRewrite(leftExpression);
+        if (left.isEmpty()) {
+            return Optional.empty();
+        }
+
+        Optional<ParameterizedExpression> right = context.defaultRewrite(rightExpression);
+        if (right.isEmpty()) {
+            return Optional.empty();
+        }
+
+        verify(expression.getFunctionName().getCatalogSchema().isEmpty()); // filtered out by the pattern
+        return Optional.of(new ParameterizedExpression(
+                format("(%s) %s (%s)", left.get().expression(), comparison.getOperator(), right.get().expression()),
+                ImmutableList.<QueryParameter>builder()
+                        .addAll(left.get().parameters())
+                        .addAll(right.get().parameters())
+                        .build()));
+    }
+
+    private static boolean isString(Variable variable, RewriteContext<?> context)
+    {
+        JdbcTypeHandle typeHandle = ((JdbcColumnHandle) context.getAssignment(variable.getName()))
+                .getJdbcTypeHandle();
+        String jdbcTypeName = typeHandle
+                .jdbcTypeName()
+                .orElseThrow(() -> new TrinoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+        return jdbcTypeName.equals("String")
```

Comment:
Instead of comparing on `jdbcTypeName` - Can we try `ClickHouseColumn.of("", jdbcTypeName)` and then use its `ClickhouseDataType` enum and decide accordingly ? This would work for both nullable and FixedString/String right ?